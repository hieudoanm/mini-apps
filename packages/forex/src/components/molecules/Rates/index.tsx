import { ForexRate } from '../../../types';
import { currencyFormatter } from '../../../utils/currency-formatter';
import List from '../List';

const Rates: React.FC<{ rates: ForexRate[] }> = ({ rates }) => {
  return (
    <List>
      <List.Header>
        <h1 className="uppercase text-center text-2xl">Forex</h1>
      </List.Header>
      {rates.map((item) => {
        const { code, currency, rate } = item;
        return (
          <List.Item key={`code-${code}`}>
            <div className="flex justify-between items-center">
              <div>
                <b>{code}</b> - {currency}
              </div>
              <div>{currencyFormatter(rate)}</div>
            </div>
          </List.Item>
        );
      })}
    </List>
  );
};

export default Rates;
