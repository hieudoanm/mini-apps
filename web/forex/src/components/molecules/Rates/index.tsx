import isNil from 'lodash/isNil';
import { vndFormatter } from '@package/shared';
import { ForexRate } from '../../../types';
import List from '../List';

const Rates: React.FC<{ rates: ForexRate[] }> = ({ rates }) => {
  return (
    <List>
      <List.Header>
        <h1 className="uppercase text-center text-2xl">Forex</h1>
      </List.Header>
      {rates.map((item) => {
        const { code, currency, rate } = item;
        return (
          <List.Item key={`code-${code}`}>
            <div className="flex justify-between items-center">
              <div>
                <span>
                  <b>{code}</b>
                </span>
                {!isNil(currency) && (
                  <span className="hidden sm:inline ml-1"> - {currency}</span>
                )}
              </div>
              <div>{vndFormatter(rate)}</div>
            </div>
          </List.Item>
        );
      })}
    </List>
  );
};

export default Rates;
