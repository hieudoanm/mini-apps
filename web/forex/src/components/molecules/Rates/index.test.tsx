import { render } from '@testing-library/react';
import Rates from '.';

describe('Rates', () => {
  test('render default', () => {
    const wrapper = render(
      <Rates rates={[{ code: 'ABC', currency: 'abcdef', rate: 1 }]} />
    );
    expect(wrapper.container).toMatchSnapshot();
  });
});
