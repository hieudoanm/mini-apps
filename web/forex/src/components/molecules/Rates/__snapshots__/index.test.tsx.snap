// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates render default 1`] = `
<div>
  <div
    class="border rounded"
  >
    <div
      class="p-4"
    >
      <h1
        class="uppercase text-center text-2xl"
      >
        Forex
      </h1>
    </div>
    <div
      class="p-4 border-t"
    >
      <div
        class="flex justify-between items-center"
      >
        <div>
          <span>
            <b>
              ABC
            </b>
          </span>
          <span
            class="hidden sm:inline ml-1"
          >
             - 
            abcdef
          </span>
        </div>
        <div>
          1â‚«
        </div>
      </div>
    </div>
  </div>
</div>
`;
